sdcm/tester.py

```diff
@@ -298,15 +298,8 @@ def __init__(self, *args):  # pylint: disable=too-many-statements
         InfoEvent('TEST_START test_id=%s' % Setup.test_id())
 
     def _init_test_timeout_thread(self) -> threading.Timer:
-        start_time = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(self.start_time))
-
         def kill_the_test():
-            try:
-                raise RuntimeError(
-                    f"Test started at {start_time} has reached timeout ({self.test_duration} minutes)"
-                )
-            except:  # pylint: disable=bare-except
-                self.kill_test(sys.exc_info())
+            TestTimeoutEvent(start_time=self.start_time, duration=self.test_duration).publish()
```

Comment:
did we consider doing both ? i.e. kill the test, but at least have the event to clearly see the reason ?

Reply:
I think that it would be good to have only one entry/exit point, meaning killing test via critical events is enough.
we need 3rd opinion here.
@bentsi , what do you think ? publishing critical and killing or just publishing critical ?

Reply:
I think it's o.k. to have a single option.
but just keep in mind this mean it would wait for nemesis to finish

the main issue I think we had with the timeouts, is that it wasn't visible enough (i.e. didn't had an event)

Reply:
I think we need one place to kill the test from.